test/src/test/java/org/corfudb/runtime/collections/UFOStoreTest.java

```diff
@@ -0,0 +1,260 @@
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StaleRevisionUpdateException;
+import org.corfudb.runtime.view.AbstractViewTest;
+import org.corfudb.test.SampleSchema;
+import org.corfudb.test.SampleSchema.EventInfo;
+import org.corfudb.test.SampleSchema.ManagedMetadata;
+import org.corfudb.test.SampleSchema.Uuid;
+import org.junit.Test;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * To ensure that feature changes in CorfuStore do not break verticals,
+ * we simulate their usage pattern with a implementation and tests.
+ *
+ * Created by hisundar on 2020-09-16
+ */
+@Slf4j
+public class UFOStoreTest extends AbstractViewTest {
+    /**
+     * UFOStoreTestImpl stores 3 pieces of information - key, value and metadata
+     * This test demonstrates how metadata field options esp "version" can be used and verified.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void checkRevisionValidation() throws Exception {
+
+        // Get a Corfu Runtime instance.
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+
+        // Creating Corfu Store using a connected corfu client.
+        UFOStoreTestImpl ufoStore = new UFOStoreTestImpl(corfuRuntime);
+
+        // Define a namespace for the table.
+        final String globalManager = "global-manager";
+        // Define table name.
+        final String tableName = "EventInfo";
+
+        // Create & Register the table.
+        // This is required to initialize the table for the current corfu client.
+        Table<Uuid, EventInfo, ManagedMetadata> table = ufoStore.openTable(
+                globalManager,
+                tableName,
+                Uuid.class,
+                EventInfo.class,
+                ManagedMetadata.class,
+                // TableOptions includes option to choose - Memory/Disk based corfu table.
+                TableOptions.builder().build());
+
+        UUID uuid1 = UUID.nameUUIDFromBytes("1".getBytes());
+        Uuid key1 = Uuid.newBuilder()
+                .setMsb(uuid1.getMostSignificantBits()).setLsb(uuid1.getLeastSignificantBits())
+                .build();
+        ManagedMetadata user_1 = ManagedMetadata.newBuilder().setCreateUser("user_1").build();
+        long expectedVersion = 0L;
+
+        ufoStore.txn(globalManager)
+                .putRecord(tableName, key1,
+                        EventInfo.newBuilder().setName("abc").build(),
+                        user_1)
+                .commit();
+
+        UFOTxnTestImpl queryTxn = ufoStore.txn(globalManager);
+        CorfuStoreEntry<Uuid, EventInfo, ManagedMetadata> entry = null;
+        try {
+            entry = queryTxn.getRecord(table, key1);
+        } catch (Throwable throwable) {
+            assertThat(throwable).isEqualTo(null);
+        } finally {
+            queryTxn.txnEnd();
```

Comment:
I think it is better to give only 1 interface for ending all types of transactions(rw, read-only and write-only).  All transactions can end using commit() is my suggestion.

Reply:
commit() is widely used to denote a change. Query is a read-only operation, that's one reason for separation.
The other reason is to start the transaction the right way - we use commit() to denote a write transaction.